docker/zenml-server-dev.Dockerfile

```diff
@@ -53,14 +53,14 @@ COPY src/zenml/__init__.py ./src/zenml/
 # NOTE: we need to uninstall zenml at the end to remove the incomplete
 # installation
 RUN pip install --upgrade pip \
-  && pip install .[server,secrets-aws,secrets-gcp,secrets-azure,secrets-hashicorp,s3fs,gcsfs,adlfs,connectors-aws,connectors-gcp,connectors-azure] \
+  && pip install .[server,secrets-aws,secrets-gcp,secrets-azure,secrets-hashicorp,s3fs,gcsfs,adlfs,connectors-aws,connectors-gcp,connectors-azure,azureml,sagemaker,vertex] \
```

Comment:
How much larger does this make the container image? When I suggested we do this, I meant doing it for ZenML Pro images only. Doing this for all server images might upset users if it adds significant bulk to the container image.

Reply:
@htahir1, @schustmi, and I had a look into this and discussed the potential outcomes. The size of our server image goes up from 242 MB to 336 MB. We all deemed it OK. What's your take on this?

Also, keep in mind, If we do it for ZenML Pro images only, we will not be able to refresh the status of the runs from the server side for our OSS users.

Reply:
I mean lets just keep it and see if people complain about the size? why should we add more complexity by splitting up the logic here

Reply:
Yeah, 90 MB increase is probably ok